prereise/gather/demanddata/eia/map_ba.py

```diff
@@ -95,3 +97,30 @@ def map_buses_to_county(bus_county_map):
         except TypeError:
             bus_no_county_match.append(index)
     return bus_county_map, bus_no_county_match
+
+
+def map_buses_to_ba(bus_gis):
+    """Find the Balancing Authority in the U.S. territory that each bus in the query grid
+    belongs to.
+
+    :param (*pandas.DataFrame*) bus_gis: data frame contains a list of
+        entries with lat and long of buses.
+    :return: (*pandas.DataFrame*) data frame of BA that buses
+        locate
+    """
+
+    b2c_key = map_buses_to_county(bus_gis)[0].iloc[:, :-1]
+
+    # read json file for BA_County Map
+    with open("../../data/BA_County_map.json") as f:
+        ba_county_map = json.load(f)
+
+    ba2county = pd.DataFrame(
+        [[k, v] for k, a in ba_county_map.items() for v in a], columns=["BA", "County"]
+    )
+
+    county_ba_map = ba2county.set_index("County").to_dict("dict")["BA"]
```

Comment:
I think in this way, we don't need `ba2county` here, right? We can simply reorganize the dict `ba_county_map` to come up with the `county_ba_map`. Then we don't need to drop last "BA" column in b2c_key data frame, we can simply do `b2c_key["BA"] = b2c_key["County"].map(county_ba_map)` or can `return b2c_key["County"].map(county_ba_map)` directly if we don't care about the rest info in the data frame, or can `return b2c_key` since we modify everything inplace.

Reply:
modified it.